### PR TITLE
Enable 15.4 packet display only

### PR DESCRIPTION
**Pull request by tbursztyka**
If more improvements are need, more features will be added then.
(i.e.: you don't want to see before/after 6lo, things like that)

### CODE REVIEW

File: subsys/net/ip/l2/ieee802154/ieee802154.c [hunk 1]
```diff
@@ -45,6 +46,12 @@ static inline void pkt_hexdump(const char *title, struct net_pkt *pkt,
 	net_hexdump_frags(title, pkt, full);
 }
 
+#ifndef CONFIG_NET_DEBUG_L2_IEEE802154
+
```
Comment (jukkar) [inline subsys/net/ip/l2/ieee802154/ieee802154.c]: Why extra line here?

File: subsys/net/ip/l2/ieee802154/ieee802154.c [hunk 2]
```diff
@@ -45,6 +46,12 @@ static inline void pkt_hexdump(const char *title, struct net_pkt *pkt,
...
+#undef NET_LOG_ENABLED
+
```
Comment (jukkar) [inline subsys/net/ip/l2/ieee802154/ieee802154.c]: and here too

Comment (tbursztyka) [reply subsys/net/ip/l2/ieee802154/ieee802154.c]: indeed, can be shrunk

**Comment by dleach02** [comment]
I agree that this is an improvement but want to point out that there is still a separation on the TX path to when a fragment/packet is actually submitted to the driver (and hardware) for transmission. In this code, the printing is done at the point that the fragment is put on the TX queue and it would be a bit more useful to have another trace/dump in the loop where the CCA/TX/retry is being doing. 

In fact, we really need a debug message (maybe INF) for failed transmit because that is something I've done in temporary code twice (once with FCSI and once with the KW41Z work).

**Review by dleach02 — APPROVED** [review]

**Review by pfalcon — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**